library/core/src/sync/atomic.rs

```diff
@@ -357,6 +357,8 @@ impl AtomicBool {
     ///     duration of lifetime `'a`. Most use cases should be able to follow this guideline.
     ///   * This requirement is also trivially satisfied if all accesses (atomic or not) are done
     ///     from the same thread.
+    /// * This method must not be used to perform overlapping or mixed-size atomic
+    ///   accesses, as these are not supported by the memory model.
```

Comment:
A bit more explanation might help here, what about this?
```suggestion
    /// * Atomic accesses to the value behind `ptr` must either all use the same size,
    ///    or they must have or they must have a happens-before relationship.
    ///   * In other words, within one of the atomic time periods mentioned in the previous
    ///      bullet, you cannot have mixed-sized or imperfectly-overlapping atomic accesses.
```

Reply:
Sounds good to me!